pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -526,7 +467,13 @@ func (c *Controller) syncCheckStates(ctx context.Context, wl *kueue.Workload, ch
 						checkState.Message = apimeta.FindStatusCondition(pr.Status.Conditions, autoscaling.Failed).Message
 					}
 				}
-			case prProvisioned:
+			case isCapacityRevoked(pr):
+				if workload.IsActive(wl) && !workload.IsFinished(wl) {
+					// happens only if the job allows retries and a user sets .spec.backOffLimit > 0
```

Comment:
I would like to explain this a little better. First, this situation may happen even if retries are not allowed if this controller reconciles before the Job controller transitions the Job to Failed. Second, there are other ways to enable retries, for example using podFailurePolicy with `ignore` action.

Maybe something like:
```
// We mark the admission check as rejected to trigger workload eviction and deactivation. 
// This is needed to prevent replacement pods being stuck in the pending phase indefinitely
// as the nodes are already deleted by Cluster Autoscaler.
```
I think something along the lines would be helpful in release notes also.